include/drivers/dma.h

```diff
@@ -224,6 +260,7 @@ __subsystem struct dma_driver_api {
 	dma_api_start start;
 	dma_api_stop stop;
 	dma_api_get_status get_status;
+	dma_chan_filter chan_filter;
```

Comment:
I'm not familiar with the story of this API but I'm a bit confused about this. `dma_chan_filter` has no `static inline` implementation in this file and apparently it is only called by the `request` call, so the only actual use of this function is internal to this API?

Reply:
yes, the scenario is that a DMA engine driver will implement a filter function and pass it to the api, and the filter logic will be called when dma_request_channel, so different DMA engine can implement different filter according its channel featues, in EDMA it is normal channel or periodic channel.

Reply:
ok, I still think that having a stub for that returning `-ENOSYS` if the driver doesn't implement it, could be a good idea.

Reply:
ok, add a stub one